src/clippoly/src/clippoly.cpp

```diff
@@ -0,0 +1,271 @@
+    p.push_back(C2::PointD(w, -h));
+    p.push_back(C2::PointD(-w, -h));
+    p.push_back(C2::PointD(-w, h));
+    p.push_back(C2::PointD(w, h));
+  } else {
+    double x = dims.x;
+    double y = dims.y;
+    p.push_back(C2::PointD(0.0, y));
+    p.push_back(C2::PointD(x, y));
+    p.push_back(C2::PointD(x, 0.0));
+    p.push_back(C2::PointD(0.0, 0.0));
+  }
+  auto ps = C2::PathsD();
+  ps.push_back(p);
+  return Clippoly(ps, true);
+}
+
+Clippoly Clippoly::Circle(float radius, int circularSegments) {
+  // GetCircularSegments(radius) -- in Manifold, not available atm
+  int n = circularSegments > 2 ? circularSegments : 3;
+  float dPhi = 360.0f / n;
+  auto circle = C2::PathsD(1);
+  for (int i = 0; i < n; ++i) {
+    circle[0].push_back(
+        C2::PointD(radius * cosd(dPhi * i), radius * sind(dPhi * i)));
+  }
+  return Clippoly(circle, true);
+}
+
+Clippoly Clippoly::Boolean(const Clippoly& second, OpType op) const {
+  auto ct = cliptype_of_op(op);
+  auto res = C2::BooleanOp(ct, C2::FillRule::NonZero, paths_, second.paths_,
+                           precision_);
+  return Clippoly(res, true);
+}
+
+Clippoly Clippoly::BatchBoolean(const std::vector<Clippoly>& clippolys,
+                                OpType op) {
+  if (clippolys.size() == 0)
+    return Clippoly();
+  else if (clippolys.size() == 1)
+    return clippolys[0];
+
+  auto subjs = clippolys[0].paths_;
+  auto clips = C2::PathsD();
+  for (int i = 1; i < clippolys.size(); ++i) {
+    auto ps = clippolys[i].paths_;
+    clips.insert(clips.end(), ps.begin(), ps.end());
+  }
+
+  auto ct = cliptype_of_op(op);
+  auto res = C2::BooleanOp(ct, C2::FillRule::NonZero, subjs, clips, precision_);
+  return Clippoly(res, true);
+}
+
+/**
+ * Shorthand for Boolean Union.
+ */
+Clippoly Clippoly::operator+(const Clippoly& Q) const {
+  return Boolean(Q, OpType::Add);
+}
+
+/**
+ * Shorthand for Boolean Union assignment.
+ */
+Clippoly& Clippoly::operator+=(const Clippoly& Q) {
+  *this = *this + Q;
+  return *this;
+}
+
+/**
+ * Shorthand for Boolean Difference.
+ */
+Clippoly Clippoly::operator-(const Clippoly& Q) const {
+  return Boolean(Q, OpType::Subtract);
+}
+
+/**
+ * Shorthand for Boolean Difference assignment.
+ */
+Clippoly& Clippoly::operator-=(const Clippoly& Q) {
+  *this = *this - Q;
+  return *this;
+}
+
+/**
+ * Shorthand for Boolean Intersection.
+ */
+Clippoly Clippoly::operator^(const Clippoly& Q) const {
+  return Boolean(Q, OpType::Intersect);
+}
+
+/**
+ * Shorthand for Boolean Intersection assignment.
+ */
+Clippoly& Clippoly::operator^=(const Clippoly& Q) {
+  *this = *this ^ Q;
+  return *this;
+}
+
+Clippoly Clippoly::Translate(glm::vec2 v) {
+  auto ps = C2::TranslatePaths(paths_, v.x, v.y);
+  return Clippoly(ps, true);
+}
+
+Clippoly Clippoly::Scale(glm::vec2 scale) {
+  auto scaled = C2::PathsD();
+  for (auto path : paths_) {
+    auto s = C2::PathD();
+    for (auto p : path) {
+      s.push_back(C2::PointD(p.x * scale.x, p.y * scale.y));
```

Comment:
Huh, they have a translate but no scale? What about rotate?

Reply:
They have scale, but it is mainly for converting between double and int64. The non-conversion one takes only a single factor for xy and the conversion scale takes a mutable int reference for as an error reporter. I just decided to skip that by doing it straight here.

Reply:
And no, no rotate.

Reply:
Let's add `Rotate` - make sure to use our `sind` and `cosd` functions so the 90-degree increments are exact.